File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    const obj = {};

    for (let i = 0; i < strs.length; i++) {
        const str = Array.from(strs[i]).sort((a, b) => a.localeCompare(b)).join('');
        if (!obj.hasOwnProperty(str)) {
            obj[str] = [];
        }
        obj[str].push(strs[i]);
    }

    return Array.from(Object.values(obj))
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        obj = {}

        for i in range(len(strs)):
            st = ''.join(sorted(list(strs[i])))
            if st not in obj:
                obj[st] = []
            obj[st].append(strs[i])
        
        return list(obj.values())